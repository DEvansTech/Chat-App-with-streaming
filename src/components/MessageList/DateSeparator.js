import React from 'react';
import styled from '@stream-io/styled-components';
import PropTypes from 'prop-types';

import { themed } from '../../styles/theme';
import { withTranslationContext } from '../../context';

const Container = styled.View`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  margin-top: 20;
  margin-bottom: 20;
  ${({ theme }) => theme.messageList.dateSeparator.container.css}
`;

const Line = styled.View`
  flex: 1;
  height: 0.5;
  background-color: ${({ theme }) => theme.colors.light};
  ${({ theme }) => theme.messageList.dateSeparator.line.css}
`;

const DateText = styled.Text`
  margin-left: 5;
  margin-right: 5;
  text-align: center;
  text-transform: uppercase;
  font-size: 10;
  opacity: 0.8;
  ${({ theme }) => theme.messageList.dateSeparator.dateText.css}
`;

const Date = styled.Text`
  font-weight: 700;
  font-size: 10;
  text-transform: uppercase;
  opacity: 0.8;
  ${({ theme }) => theme.messageList.dateSeparator.date.css}
`;

/**
 * @extends PureComponent
 * @example ../docs/DateSeparator.md
 */

class DateSeparator extends React.PureComponent {
  static propTypes = {
    message: PropTypes.object.isRequired,
    /**
     * Formatter function for date object.
     *
     * @param date Date object of message
     * @returns string
     */
    formatDate: PropTypes.func,
  };

  static themePath = 'messageList.dateSeparator';

  render() {
    const { message, formatDate, tDateTimeParser } = this.props;

    return (
      <Container>
        <Line />
        <DateText>
          {formatDate ? (
            formatDate(message.date)
          ) : (
            <React.Fragment>
              <Date>{tDateTimeParser(message.date).calendar()}</Date>
            </React.Fragment>
          )}
        </DateText>
        <Line />
      </Container>
    );
  }
}

export default withTranslationContext(themed(DateSeparator));
