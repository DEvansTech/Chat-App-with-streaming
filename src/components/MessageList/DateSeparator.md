The date separator between messages.
Here's what it looks like for today.

```js
const date = new Date();
const data = require('../docs/data');

<Chat client={data.client}>
  <DateSeparator {...data.translationContext} message={{ date }} />
  <DateSeparator
    {...data.translationContext}
    message={{ date }}
    alignment="center"
  />
  <DateSeparator
    {...data.translationContext}
    message={{ date }}
    alignment="left"
  />
</Chat>;
```

and for a date in the past:

```js
const data = require('../docs/data');

const date = new Date('December 17, 1995 03:24:00');
<Chat client={data.client}>
  <DateSeparator {...data.translationContext} message={{ date }} />
  <DateSeparator
    {...data.translationContext}
    message={{ date }}
    alignment="center"
  />
  <DateSeparator
    {...data.translationContext}
    message={{ date }}
    alignment="left"
  />
</Chat>;
```

and adding custom date formatting:

```js
const data = require('../docs/data');

const date = new Date('December 17, 1995 03:24:00');

function formatDate(d) {
  return <h2>Messages posted after {d.toDateString()}</h2>;
}

<Chat client={data.client}>
  <DateSeparator
    {...data.translationContext}
    formatDate={formatDate}
    message={{ date }}
  />
  <DateSeparator
    {...data.translationContext}
    formatDate={formatDate}
    message={{ date }}
    alignment="center"
  />
  <DateSeparator
    {...data.translationContext}
    formatDate={formatDate}
    message={{ date }}
    alignment="left"
  />
</Chat>;
```
