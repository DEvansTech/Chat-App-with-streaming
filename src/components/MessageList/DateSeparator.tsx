import React from 'react';

import type { DateSeparator as DateSeparatorType } from './utils/insertDates';

import {
  isDayOrMoment,
  TDateTimeParserInput,
  useTranslationContext,
} from '../../contexts/translationContext/TranslationContext';
import { styled } from '../../styles/styledComponents';
import { themed } from '../../styles/theme';

import type {
  DefaultAttachmentType,
  DefaultChannelType,
  DefaultCommandType,
  DefaultEventType,
  DefaultMessageType,
  DefaultReactionType,
  DefaultUserType,
} from '../../types/types';

const Container = styled.View`
  align-items: center;
  flex-direction: row;
  justify-content: center;
  margin-vertical: 20px;
  ${({ theme }) => theme.messageList.dateSeparator.container.css}
`;

const Date = styled.Text`
  font-size: 10px;
  font-weight: 700;
  opacity: 0.8;
  text-transform: uppercase;
  ${({ theme }) => theme.messageList.dateSeparator.date.css}
`;

const DateText = styled.Text`
  font-size: 10px;
  margin-horizontal: 5px;
  opacity: 0.8;
  text-align: center;
  text-transform: uppercase;
  ${({ theme }) => theme.messageList.dateSeparator.dateText.css}
`;

const Line = styled.View`
  background-color: ${({ theme }) => theme.colors.light};
  flex: 1;
  height: 0.5px;
  ${({ theme }) => theme.messageList.dateSeparator.line.css}
`;

export type DateSeparatorProps<
  At extends Record<string, unknown> = DefaultAttachmentType,
  Ch extends Record<string, unknown> = DefaultChannelType,
  Co extends string = DefaultCommandType,
  Ev extends Record<string, unknown> = DefaultEventType,
  Me extends Record<string, unknown> = DefaultMessageType,
  Re extends Record<string, unknown> = DefaultReactionType,
  Us extends Record<string, unknown> = DefaultUserType
> = {
  message: DateSeparatorType<At, Ch, Co, Ev, Me, Re, Us>;
  /**
   * Formatter function for date object.
   *
   * @param date TDateTimeParserInput object of message
   * @returns string
   */
  formatDate?: (date: TDateTimeParserInput) => string;
};

/**
 * @example ./DateSeparator.md
 */
const DateSeparator = <
  At extends Record<string, unknown> = DefaultAttachmentType,
  Ch extends Record<string, unknown> = DefaultChannelType,
  Co extends string = DefaultCommandType,
  Ev extends Record<string, unknown> = DefaultEventType,
  Me extends Record<string, unknown> = DefaultMessageType,
  Re extends Record<string, unknown> = DefaultReactionType,
  Us extends Record<string, unknown> = DefaultUserType
>({
  formatDate,
  message,
}: DateSeparatorProps<At, Ch, Co, Ev, Me, Re, Us>) => {
  const { tDateTimeParser } = useTranslationContext();

  let date;

  if (formatDate) {
    date = formatDate(message.date as TDateTimeParserInput);
  } else {
    date = tDateTimeParser(message.date as TDateTimeParserInput);
  }

  return (
    <Container testID='date-separator'>
      <Line />
      <DateText>
        {formatDate ? (
          date
        ) : (
          <Date>{isDayOrMoment(date) ? date.calendar() : date}</Date>
        )}
      </DateText>
      <Line />
    </Container>
  );
};

DateSeparator.themePath = 'messageList.dateSeparator';

export default themed(DateSeparator) as typeof DateSeparator;
