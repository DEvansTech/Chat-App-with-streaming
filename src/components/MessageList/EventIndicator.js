import React from 'react';
import styled from '@stream-io/styled-components';
import PropTypes from 'prop-types';

import { withTranslationContext } from '../../context';
import { Avatar } from '../Avatar';

const Date = styled.Text`
  font-size: 10;
  color: rgba(0, 0, 0, 0.5);
  ${({ theme }) => theme.messageList.eventIndicator.date.css}
`;

const MemberUpdateContainer = styled.View`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  padding: 10px;
  ${({ theme }) => theme.messageList.eventIndicator.memberUpdateContainer.css}
`;

const MemberUpdateTextContainer = styled.View`
  display: flex;
  flex-direction: column;
  margin-left: 10px;
  ${({ theme }) =>
    theme.messageList.eventIndicator.memberUpdateTextContainer.css}
`;

const MemberUpdateText = styled.Text`
  font-size: 13px;
  font-style: italic;
  color: rgba(0, 0, 0, 0.5);
  ${({ theme }) => theme.messageList.eventIndicator.memberUpdateText.css}
`;

/**
 * A component to display a message regarding channel notifications such as
 * 'member.added', 'member.removed' etc.
 */
const EventIndicator = ({ event, t, tDateTimeParser }) => {
  if (event.type === 'member.added' || event.type === 'member.removed') {
    return (
      <MemberUpdateContainer>
        <Avatar name={event.user.name} image={event.user.image} />
        <MemberUpdateTextContainer>
          <MemberUpdateText>
            {event.type === 'member.added'
              ? t('{{ username }} joined the chat', {
                  username: event.user.name,
                })
              : t('{{ username }} was removed from the chat', {
                  username: event.user.name,
                })}
          </MemberUpdateText>
          <Date>{tDateTimeParser(event.received_at).format('LT')}</Date>
        </MemberUpdateTextContainer>
      </MemberUpdateContainer>
    );
  }

  return null;
};

EventIndicator.propTypes = {
  event: PropTypes.object,
};

export default withTranslationContext(EventIndicator);
