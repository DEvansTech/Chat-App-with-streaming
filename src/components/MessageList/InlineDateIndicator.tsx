import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

import { useTheme } from '../../contexts/themeContext/ThemeContext';
import {
  isDayOrMoment,
  useTranslationContext,
} from '../../contexts/translationContext/TranslationContext';

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    alignSelf: 'center',
    borderRadius: 8,
    height: 16,
    justifyContent: 'center',
    marginVertical: 4,
    paddingHorizontal: 8,
  },
  text: {
    fontSize: 12,
    textAlign: 'center',
    textAlignVertical: 'center',
  },
});

export type InlineDateIndicatorProps = {
  date?: Date;
};

export const InlineDateIndicator: React.FC<InlineDateIndicatorProps> = ({
  date,
}) => {
  const {
    theme: {
      colors: { overlay_dark, white },
      inlineDateIndicator: { container, text },
    },
  } = useTheme();
  const { tDateTimeParser } = useTranslationContext();

  if (!date) {
    return null;
  }

  const dateFormat =
    date.getFullYear() === new Date().getFullYear() ? 'MMM D' : 'MMM D, YYYY';
  const tDate = tDateTimeParser(date);
  const dateString = isDayOrMoment(tDate)
    ? tDate.format(dateFormat)
    : new Date(tDate).toDateString();

  return (
    <View
      style={[styles.container, { backgroundColor: overlay_dark }, container]}
    >
      <Text style={[styles.text, { color: white }, text]}>{dateString}</Text>
    </View>
  );
};
