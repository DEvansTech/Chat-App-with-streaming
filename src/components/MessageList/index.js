export { default as DateSeparator } from './DateSeparator';
export { default as EventIndicator } from './EventIndicator';
export { default as MessageList } from './MessageList';
export { default as MessageNotification } from './MessageNotification';
export { default as MessageSystem } from './MessageSystem';
export { default as TypingIndicator } from './TypingIndicator';
