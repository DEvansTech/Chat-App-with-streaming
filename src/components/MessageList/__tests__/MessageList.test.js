import React from 'react';
import { cleanup, render, waitFor } from '@testing-library/react-native';

import {
  useMockedApis,
  getOrCreateChannelApi,
  generateChannel,
  generateMessage,
  generateStaticMessage,
  generateMember,
  generateStaticUser,
  generateUser,
  dispatchMessageNewEvent,
  getTestClientWithUser,
} from 'mock-builders';

import { Chat } from '../../Chat';
import MessageList from '../MessageList';
import { Channel } from '../../Channel';

describe('MessageList', () => {
  afterEach(cleanup);

  let chatClient;

  it('should add new message at bottom of the list', async () => {
    const user1 = generateUser();
    const user2 = generateUser();
    const mockedChannel = generateChannel({
      messages: [
        generateMessage({ user: user1 }),
        generateMessage({ user: user1 }),
      ],
      members: [
        generateMember({ user: user1 }),
        generateMember({ user: user1 }),
      ],
    });

    chatClient = await getTestClientWithUser({ id: 'testID' });
    useMockedApis(chatClient, [getOrCreateChannelApi(mockedChannel)]);
    const channel = chatClient.channel('messaging', mockedChannel.id);
    await channel.query();

    const { getByText } = render(
      <Chat client={chatClient}>
        <Channel channel={channel}>
          <MessageList />
        </Channel>
      </Chat>,
    );

    const newMessage = generateMessage({ user: user2 });
    dispatchMessageNewEvent(chatClient, newMessage, mockedChannel.channel);

    await waitFor(() => {
      expect(getByText(newMessage.text)).toBeTruthy();
    });
  });

  it('should render the message list', async () => {
    const user1 = generateStaticUser(1);
    const user2 = generateStaticUser(2);
    const mockedChannel = generateChannel({
      messages: [
        generateStaticMessage(
          'Message1',
          { user: user1 },
          '2020-05-05T14:48:00',
        ),
        generateStaticMessage(
          'Message2',
          { user: user2 },
          '2020-05-05T14:49:00',
        ),
        generateStaticMessage(
          'Message3',
          { user: user2 },
          '2020-05-05T14:50:00',
        ),
      ],
      members: [
        generateMember({ user: user1 }),
        generateMember({ user: user1 }),
      ],
    });

    chatClient = await getTestClientWithUser({ id: 'testID2' });
    useMockedApis(chatClient, [getOrCreateChannelApi(mockedChannel)]);
    const channel = chatClient.channel('messaging', mockedChannel.id);
    await channel.query();

    const { toJSON } = render(
      <Chat client={chatClient}>
        <Channel channel={channel}>
          <MessageList />
        </Channel>
      </Chat>,
    );

    await waitFor(() => {
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
