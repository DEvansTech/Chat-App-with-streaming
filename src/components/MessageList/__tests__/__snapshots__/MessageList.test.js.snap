// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList should render the message list and match snapshot 1`] = `
<View
  keyboardVerticalOffset={86.5}
  onLayout={[Function]}
  style={
    Object {
      "paddingBottom": 0,
    }
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "width": "100%",
        }
      }
    >
      <RCTScrollView
        data={
          Array [
            Object {
              "__html": "<p>regular</p>",
              "attachments": Array [],
              "created_at": 2020-05-06T14:50:00.000Z,
              "groupStyles": Array [
                "single",
              ],
              "html": "<p>regular</p>",
              "id": "b4612a73-fa2b-5787-bf71-1adc8f291a04",
              "readBy": true,
              "status": "received",
              "text": "Message3",
              "type": "regular",
              "updated_at": 2020-05-06T14:50:00.000Z,
              "user": Object {
                "banned": false,
                "created_at": "2020-04-27T13:39:49.331742Z",
                "id": "ViriBoi2",
                "image": "https://i.imgur.com/iNaC3K7.jpg",
                "name": "Vir",
                "online": false,
                "role": "user",
                "updated_at": "2020-04-27T13:39:49.332087Z",
              },
            },
            Object {
              "date": 2020-05-06T14:50:00.000Z,
              "groupStyles": Array [],
              "readBy": false,
              "type": "message.date",
            },
            Object {
              "__html": "<p>regular</p>",
              "attachments": Array [],
              "created_at": 2020-05-05T14:49:00.000Z,
              "groupStyles": Array [
                "single",
              ],
              "html": "<p>regular</p>",
              "id": "b7e00d73-82c4-59b6-a465-0785af2bc6d9",
              "readBy": true,
              "status": "received",
              "text": "Message2",
              "type": "regular",
              "updated_at": 2020-05-05T14:49:00.000Z,
              "user": Object {
                "banned": false,
                "created_at": "2020-04-27T13:39:49.331742Z",
                "id": "ViriBoi2",
                "image": "https://i.imgur.com/iNaC3K7.jpg",
                "name": "Vir",
                "online": false,
                "role": "user",
                "updated_at": "2020-04-27T13:39:49.332087Z",
              },
            },
            Object {
              "__html": "<p>regular</p>",
              "attachments": Array [],
              "created_at": 2020-05-05T14:48:00.000Z,
              "groupStyles": Array [
                "single",
              ],
              "html": "<p>regular</p>",
              "id": "e06aaa16-3deb-5d91-bca5-50efc733c183",
              "readBy": true,
              "status": "received",
              "text": "Message1",
              "type": "regular",
              "updated_at": 2020-05-05T14:48:00.000Z,
              "user": Object {
                "banned": false,
                "created_at": "2020-04-27T13:39:49.331742Z",
                "id": "GrantyBoi2",
                "image": "https://i.imgur.com/SLx06PP.png",
                "name": "Grant",
                "online": false,
                "role": "user",
                "updated_at": "2020-04-27T13:39:49.332087Z",
              },
            },
            Object {
              "date": 2020-05-05T14:48:00.000Z,
              "groupStyles": Array [],
              "readBy": false,
              "type": "message.date",
            },
          ]
        }
        disableVirtualization={false}
        extraData={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        invertStickyHeaders={true}
        inverted={true}
        keyExtractor={[Function]}
        keyboardShouldPersistTaps="handled"
        maintainVisibleContentPosition={
          Object {
            "autoscrollToTopThreshold": 10,
            "minIndexForVisible": 1,
          }
        }
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReached={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        style={
          Array [
            Object {
              "transform": Array [
                Object {
                  "scaleY": -1,
                },
              ],
            },
            Array [
              Object {
                "flex": 1,
                "paddingHorizontal": 10,
                "width": "100%",
              },
              Object {},
            ],
          ]
        }
        testID="message-flat-list"
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "column-reverse",
                },
                Object {
                  "transform": Array [
                    Object {
                      "scaleY": -1,
                    },
                  ],
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "flex-end",
                    "flexDirection": "row",
                  },
                  Object {
                    "justifyContent": "flex-start",
                    "marginBottom": 30,
                    "marginTop": 0,
                  },
                  Object {},
                ]
              }
              testID="message-simple-wrapper"
            >
              <View
                style={
                  Array [
                    Object {
                      "marginRight": 8,
                    },
                    Object {},
                  ]
                }
                testID="message-avatar"
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                      },
                      Object {},
                      undefined,
                    ]
                  }
                >
                  <Image
                    accessibilityLabel="initials"
                    onError={[Function]}
                    resizeMethod="resize"
                    source={
                      Object {
                        "uri": "https://i.imgur.com/iNaC3K7.jpg",
                      }
                    }
                    style={
                      Array [
                        Object {
                          "borderRadius": 16,
                          "height": 32,
                          "width": 32,
                        },
                        Object {},
                      ]
                    }
                    testID="avatar-image"
                  />
                </View>
              </View>
              <View
                accessible={true}
                focusable={false}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "flex-start",
                    "borderBottomLeftRadius": 2,
                    "borderBottomRightRadius": 16,
                    "borderTopLeftRadius": 16,
                    "borderTopRightRadius": 16,
                    "justifyContent": "flex-start",
                    "opacity": 1,
                  }
                }
              >
                <View
                  onLayout={[Function]}
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "flex-start",
                        },
                        Object {},
                      ]
                    }
                    testID="message-content-wrapper"
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "marginTop": 2,
                            "maxWidth": 250,
                            "paddingHorizontal": 16,
                          },
                          Object {
                            "alignSelf": "flex-start",
                            "borderColor": "#00000014",
                            "borderWidth": 0.5,
                          },
                          Object {
                            "backgroundColor": "#FFFFFF",
                            "borderBottomLeftRadius": 2,
                            "borderBottomRightRadius": 16,
                            "borderTopLeftRadius": 16,
                            "borderTopRightRadius": 16,
                          },
                          Object {},
                        ]
                      }
                      testID="message-text-container"
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "alignSelf": "stretch",
                            },
                            undefined,
                          ]
                        }
                      >
                        <Text
                          style={
                            Object {
                              "alignItems": "flex-start",
                              "flexDirection": "row",
                              "flexWrap": "wrap",
                              "justifyContent": "flex-start",
                              "marginBottom": 0,
                              "marginTop": 0,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "#222222",
                              }
                            }
                          >
                            Message3
                          </Text>
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "marginTop": 4,
                    }
                  }
                  testID="message-status-time"
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "textAlign": "left",
                        },
                        Object {
                          "color": "#00000080",
                          "fontSize": 12,
                        },
                      ]
                    }
                  >
                    2:50 PM
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "column-reverse",
                },
                Object {
                  "transform": Array [
                    Object {
                      "scaleY": -1,
                    },
                  ],
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginVertical": 20,
                  },
                  Object {},
                ]
              }
              testID="date-separator"
            >
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "height": 0.5,
                    },
                    Object {
                      "backgroundColor": "#EBEBEB",
                    },
                    Object {},
                  ]
                }
              />
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 10,
                      "marginHorizontal": 5,
                      "opacity": 0.8,
                      "textAlign": "center",
                      "textTransform": "uppercase",
                    },
                    Object {},
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontSize": 10,
                        "fontWeight": "700",
                        "opacity": 0.8,
                        "textTransform": "uppercase",
                      },
                      Object {},
                    ]
                  }
                >
                  05/06/2020
                </Text>
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "height": 0.5,
                    },
                    Object {
                      "backgroundColor": "#EBEBEB",
                    },
                    Object {},
                  ]
                }
              />
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "column-reverse",
                },
                Object {
                  "transform": Array [
                    Object {
                      "scaleY": -1,
                    },
                  ],
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "flex-end",
                    "flexDirection": "row",
                  },
                  Object {
                    "justifyContent": "flex-start",
                    "marginBottom": 8,
                    "marginTop": 0,
                  },
                  Object {},
                ]
              }
              testID="message-simple-wrapper"
            >
              <View
                style={
                  Array [
                    Object {
                      "marginRight": 8,
                    },
                    Object {},
                  ]
                }
                testID="message-avatar"
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                      },
                      Object {},
                      undefined,
                    ]
                  }
                >
                  <Image
                    accessibilityLabel="initials"
                    onError={[Function]}
                    resizeMethod="resize"
                    source={
                      Object {
                        "uri": "https://i.imgur.com/iNaC3K7.jpg",
                      }
                    }
                    style={
                      Array [
                        Object {
                          "borderRadius": 16,
                          "height": 32,
                          "width": 32,
                        },
                        Object {},
                      ]
                    }
                    testID="avatar-image"
                  />
                </View>
              </View>
              <View
                accessible={true}
                focusable={false}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "flex-start",
                    "borderBottomLeftRadius": 2,
                    "borderBottomRightRadius": 16,
                    "borderTopLeftRadius": 16,
                    "borderTopRightRadius": 16,
                    "justifyContent": "flex-start",
                    "opacity": 1,
                  }
                }
              >
                <View
                  onLayout={[Function]}
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "flex-start",
                        },
                        Object {},
                      ]
                    }
                    testID="message-content-wrapper"
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "marginTop": 2,
                            "maxWidth": 250,
                            "paddingHorizontal": 16,
                          },
                          Object {
                            "alignSelf": "flex-start",
                            "borderColor": "#00000014",
                            "borderWidth": 0.5,
                          },
                          Object {
                            "backgroundColor": "#FFFFFF",
                            "borderBottomLeftRadius": 2,
                            "borderBottomRightRadius": 16,
                            "borderTopLeftRadius": 16,
                            "borderTopRightRadius": 16,
                          },
                          Object {},
                        ]
                      }
                      testID="message-text-container"
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "alignSelf": "stretch",
                            },
                            undefined,
                          ]
                        }
                      >
                        <Text
                          style={
                            Object {
                              "alignItems": "flex-start",
                              "flexDirection": "row",
                              "flexWrap": "wrap",
                              "justifyContent": "flex-start",
                              "marginBottom": 0,
                              "marginTop": 0,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "#222222",
                              }
                            }
                          >
                            Message2
                          </Text>
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "marginTop": 4,
                    }
                  }
                  testID="message-status-time"
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "textAlign": "left",
                        },
                        Object {
                          "color": "#00000080",
                          "fontSize": 12,
                        },
                      ]
                    }
                  >
                    2:49 PM
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "column-reverse",
                },
                Object {
                  "transform": Array [
                    Object {
                      "scaleY": -1,
                    },
                  ],
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "flex-end",
                    "flexDirection": "row",
                  },
                  Object {
                    "justifyContent": "flex-start",
                    "marginBottom": 8,
                    "marginTop": 0,
                  },
                  Object {},
                ]
              }
              testID="message-simple-wrapper"
            >
              <View
                style={
                  Array [
                    Object {
                      "marginRight": 8,
                    },
                    Object {},
                  ]
                }
                testID="message-avatar"
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                      },
                      Object {},
                      undefined,
                    ]
                  }
                >
                  <Image
                    accessibilityLabel="initials"
                    onError={[Function]}
                    resizeMethod="resize"
                    source={
                      Object {
                        "uri": "https://i.imgur.com/SLx06PP.png",
                      }
                    }
                    style={
                      Array [
                        Object {
                          "borderRadius": 16,
                          "height": 32,
                          "width": 32,
                        },
                        Object {},
                      ]
                    }
                    testID="avatar-image"
                  />
                </View>
              </View>
              <View
                accessible={true}
                focusable={false}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "flex-start",
                    "borderBottomLeftRadius": 2,
                    "borderBottomRightRadius": 16,
                    "borderTopLeftRadius": 16,
                    "borderTopRightRadius": 16,
                    "justifyContent": "flex-start",
                    "opacity": 1,
                  }
                }
              >
                <View
                  onLayout={[Function]}
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "flex-start",
                        },
                        Object {},
                      ]
                    }
                    testID="message-content-wrapper"
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "marginTop": 2,
                            "maxWidth": 250,
                            "paddingHorizontal": 16,
                          },
                          Object {
                            "alignSelf": "flex-start",
                            "borderColor": "#00000014",
                            "borderWidth": 0.5,
                          },
                          Object {
                            "backgroundColor": "#FFFFFF",
                            "borderBottomLeftRadius": 2,
                            "borderBottomRightRadius": 16,
                            "borderTopLeftRadius": 16,
                            "borderTopRightRadius": 16,
                          },
                          Object {},
                        ]
                      }
                      testID="message-text-container"
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "alignSelf": "stretch",
                            },
                            undefined,
                          ]
                        }
                      >
                        <Text
                          style={
                            Object {
                              "alignItems": "flex-start",
                              "flexDirection": "row",
                              "flexWrap": "wrap",
                              "justifyContent": "flex-start",
                              "marginBottom": 0,
                              "marginTop": 0,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "#222222",
                              }
                            }
                          >
                            Message1
                          </Text>
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "marginTop": 4,
                    }
                  }
                  testID="message-status-time"
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "textAlign": "left",
                        },
                        Object {
                          "color": "#00000080",
                          "fontSize": 12,
                        },
                      ]
                    }
                  >
                    2:48 PM
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "column-reverse",
                },
                Object {
                  "transform": Array [
                    Object {
                      "scaleY": -1,
                    },
                  ],
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginVertical": 20,
                  },
                  Object {},
                ]
              }
              testID="date-separator"
            >
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "height": 0.5,
                    },
                    Object {
                      "backgroundColor": "#EBEBEB",
                    },
                    Object {},
                  ]
                }
              />
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 10,
                      "marginHorizontal": 5,
                      "opacity": 0.8,
                      "textAlign": "center",
                      "textTransform": "uppercase",
                    },
                    Object {},
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontSize": 10,
                        "fontWeight": "700",
                        "opacity": 0.8,
                        "textTransform": "uppercase",
                      },
                      Object {},
                    ]
                  }
                >
                  05/05/2020
                </Text>
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "height": 0.5,
                    },
                    Object {
                      "backgroundColor": "#EBEBEB",
                    },
                    Object {},
                  ]
                }
              />
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
